tests/e2e/pageObjects/Viewer.ts

```diff
@@ -0,0 +1,120 @@
+import { Page, expect } from '@playwright/test'
+import { state } from '../hooks'
+import util from 'util'
+
+export class Viewer {
+  page: Page = state.page
+  elements: Readonly<Record<string, string>> = {
+    resourceNameSelector: '#files-space-table [data-test-resource-name="%s"]',
+    appbarResourceNameSelector: '#app-top-bar-resource [data-test-resource-name="%s"]',
+    appTopBar: '.oc-app-top-bar .oc-resource', // '.oc-app-top-bar',
+    appTopBarResourceName: '.oc-resource-name',
+    modelViewport: '#preview .model-viewport',
+    modelViewportWrapper: '#preview #scene-wrapper',
+    modelViewportWrapperFullscreen: '#scene-wrapper:fullscreen',
+    modelViewportDescription: '#preview h1.oc-invisible-sr',
+    modelViewportCanvas: '#preview .model-viewport canvas',
+    controlButtonPrev: '.preview-controls-previous',
+    controlButtonNext: '.preview-controls-next',
+    controlButtonFullscreen: '.preview-controls-fullscreen',
+    controlButtonReset: '. preview-controls-reset'
+  }
+
+  async previewFile(filename: string): Promise<void> {
+    await this.page.locator(util.format(this.elements.resourceNameSelector, filename)).click()
+    await expect(this.page.locator(this.elements.modelViewport)).toBeVisible()
+  }
+
+  async checkViewport(filename: string): Promise<void> {
+    await expect(this.page.locator(this.elements.modelViewport)).toBeVisible()
+    await expect(this.page.locator(this.elements.modelViewportCanvas)).toBeVisible()
+    // check if file name is displayed in hidden h1 title element
+    const description = await this.page.locator(this.elements.modelViewportDescription).innerText()
+    await expect(description).toContain(filename)
```

Comment:
In my opinion the assertions should be in the Context not the page-object.
The page-object is a representation of the page and can return anything that is shown on the page or do actions on the page. The context should check if the results are correct

Reply:
@individual-it thanks for the feedback. I did some refactoring. Would you mind to have another look?

PS: steps regarding model modification by mouse interaction (zoom, rotation) are not yet implemented. please ignore them for the time being.